### PR TITLE
[CircleViewer] Tidy registration

### PR DESCRIPTION
**Pull request by dayo09**
This commit tidies registration of CircleViewerProvider.

ONE-vscode-DCO-1.0-Signed-off-by: dayo09 <dayoung.lee@samsung.com>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export function activate(context: vscode.ExtensionContext) {
   context.subscriptions.push(PartEditorProvider.register(context));
   context.subscriptions.push(PartGraphSelPanel.register(context));
 
-  context.subscriptions.push(CircleViewerProvider.register(context));
+  CircleViewerProvider.register(context);
```
Comment (dayo09) [inline src/extension.ts:77]: From [#1167](https://github.com/Samsung/ONE-vscode/issues/1167#issue-1329445337),

The function is tidy as is, but I am trying to standardize the registration function as below. 😄 

> Point 1. introduce `public static register(context: vscode.ExtensionContext) : void` function
> Point 2. Register each modules with the static function call
> ```
> OneTreeDataProvider.register(context);
> ```

File: src/CircleGraph/CircleViewer.ts [hunk 2]
```diff
@@ -103,17 +103,19 @@ export class CircleViewerProvider implements
 
   private _context: vscode.ExtensionContext;
 
-  public static register(context: vscode.ExtensionContext): vscode.Disposable {
```
Comment (dayo09) [inline src/CircleGraph/CircleViewer.ts:106]: From [#1167](https://github.com/Samsung/ONE-vscode/issues/1167#issue-1329445337),

The function is tidy as is, but I am trying to standardize the registration function as below. 😄 

> Point 1. introduce `public static register(context: vscode.ExtensionContext) : void` function
> Point 2. Register each modules with the static function call
> ```
> OneTreeDataProvider.register(context);
> ```

File: src/extension.ts (hunk 1 above)
Comment (dayo09) [inline src/extension.ts:77]: A little bit of improvement is that it does registration at one place. (not passing the provider disposable to the caller)

File: src/CircleGraph/CircleViewer.ts [hunk 3]
```diff
@@ -103,17 +103,19 @@ export class CircleViewerProvider implements
...
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(CircleViewerProvider.viewType, provider, {
+        webviewOptions: {
+          retainContextWhenHidden: true,
+        },
+        supportsMultipleEditorsPerDocument: true,
+      })
+      // Add command registration here
+    ];
+    registrations.forEach(disposable => context.subscriptions.push(disposable));
```
Comment (seanshpark) [inline src/CircleGraph/CircleViewer.ts:118]: why do you put this in array?

Comment (seanshpark) [reply src/CircleGraph/CircleViewer.ts:118]: I'v found answer in PartEditor register.

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**